src/table/components/head/HeadCellMenu.tsx

```diff
@@ -35,24 +47,28 @@ export default function HeadCellMenu({ columnIndex, isDimension, tabIndex }: Hea
           ]
         : []),
     ],
-    [translator, isDimension]
+    [translator, isDimension, isMasterDimension]
   );
 
   return menuItemGroups.length ? (
     <>
-      <StyledMenuIconButton
-        isVisible={openListboxDropdown || openMenuDropdown}
-        ref={anchorRef}
-        size="small"
-        tabIndex={tabIndex}
-        id="sn-table-head-menu-button"
-        aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
-        aria-expanded={openMenuDropdown ? 'true' : undefined}
-        aria-haspopup="true"
-        onClick={() => setOpenMenuDropdown(!openMenuDropdown)}
-      >
-        <More height="12px" />
-      </StyledMenuIconButton>
+      <div>
+        <StyledMenuIconButton
+          isVisible={openMenuDropdown}
+          ref={anchorRef}
+          size="small"
+          tabIndex={tabIndex}
+          id="sn-table-head-menu-button"
+          aria-controls={openMenuDropdown ? 'sn-table-head-menu' : undefined}
+          aria-expanded={openMenuDropdown ? 'true' : undefined}
+          aria-haspopup="true"
+          onClick={() => setOpenMenuDropdown(!openMenuDropdown)}
+        >
+          <More size="small" />
+        </StyledMenuIconButton>
+
+        <div ref={listboxRef} />
```

Comment:
why does this need to be wrapped in a div?

Reply:
because the parent element is flex and we only want to show 2 things in a header cell (title button +  '...' menu ). if we put it without a wrapper div, we will end up showing 3 flexed items stretching over head cell.

Reply:
Ah I see the popover itself is not added here in the tree, but it is mounted on this div, and that makes the alignment in the flexbox strange?

Reply:
not actually, this small div is acting like an anchor point for popover. That is the only use case for it. but because of existence of this div directly in HeadCellMenu while it has a display:flex -> it makes it's direct children to spread within it. so wrapping it inside of a div avoids that problem.